napari/layers/utils/layer_utils.py

```diff
@@ -278,7 +300,312 @@ def calc_data_range(data, rgb=False) -> Tuple[float, float]:
+
+    return slices
+
+
+def _get_blocks_grid_shape(
+    data_shape: Sequence[int], chunk_size: Sequence[int]
+) -> Tuple[int, ...]:
+    """
+    Get the approximate shape of the grid of array chunks.
+
+    Gets the shape of the array not on the pixel level but on the chunk level. For example if x is 100 and
+    chunk size is 10, 10 chunks fit in 100 and will thus be returned. In case of x being 105, still 10 will be
+    returned. This to ensure that only whole chunks will be loaded.
+
+    Parameters
+    ----------
+    data_shape: tuple[int, ...]
+        The shape of an array of chunked data.
+    chunk_size: tuple[int, ...]
+        The size per dimension of the chunks in the chunked data array.
+
+    Returns
+    -------
+    tuple[int, ...]
+        shape indicating the number of chunks per dimension.
+    """
+    return tuple(
+        data_shape[i] // chunk_size[i] for i in range(len(data_shape))
+    )
+
+
+def _get_plane_indices(
+    shape: Sequence[int], offset: int
+) -> List[Tuple[int, ...]]:
+    """
+    Get the indices that correspond to the lowest, middle and highest index of the non-visible dimensions in shape.
+
+    In case the different planes are the same, duplicates are removed.
+
+    Parameters
+    ----------
+    shape: Sequence[int]
+        Either the shape of the raw data (in pixels) or the block shape of the data (chunks)
+    offset: int
+        Number of visible dimensions
+
+    Returns
+    -------
+    idxs: list[tuple[int, ...]]
+        Bottom, middle and top plane for each non-visible dimension or single plane
+    """
+    bottom_plane_idx = (0,) * (len(shape) - offset)
+    middle_plane_idx = tuple(s // 2 for s in shape[:-offset])
+    top_plane_idx = tuple(s - 1 for s in shape[:-offset])
+    idxs = [bottom_plane_idx, middle_plane_idx, top_plane_idx]
+    if len(idxs) != len(set(idxs)):
+        return list(set(idxs))
+    return idxs
+
+
+def _get_crop_slices(
+    shape: Sequence[int],
+    plane_indices: Sequence[Sequence[int]],
+    offset: int,
+    chunk_shape: None | Tuple[int, ...] = None,
+) -> Union[None, List[Tuple[Union[int, slice], ...]]]:
+    """
+    Get the crop slices to be used for determining contrast limits when data is larger than the pixel threshold.
+
+    Parameters
+    ----------
+    shape: Sequence[int]
+        Either the shape of the raw data (in pixels) or the block shape of the data (chunks)
+    plane_indices: Sequence[Sequence[int]]
+        Bottom, middle and top plane or single plane index for each non-visible dimension.
+    offset: int
+        Number of visible dimensions.
+    chunk_shape: None | Tuple[int]
+        The size per dimension of the chunks.
+
+    Returns
+    -------
+    slices: Union[None, List[Tuple[Union[int, slice], ...]]]
+        A list of crop slices.
+    """
+    plane_shape = shape[-offset:]
+
+    # defaults in case we are dealing with numpy array. If not these will be overwritten.
+    max_chunks_per_plane = 9
```

Comment:
It should be a global variable.

Reply:
I have changed the variable name and added to docstring to be more clear as to why this is set to 9. Since we sample at the quartiles this is always 9 and we purely use it for calculation purposes in case the chunk size do not allow us to sample 9 crops.

Reply:
reverted variable name, but still clarified in docstring why this is actually always 9 (per 1st 2nd and 3rd quartile for both y and x dimension).